UnitTestFrameworkPkg/Test/GoogleTest/Sample/SampleGoogleTest/SampleGoogleTest.cpp

```diff
@@ -294,12 +295,151 @@ TEST (MacroTestsMessages, MacroTraceMessage) {
+  ASSERT_EQ (Value, Value);
+
+  FreePool (Pointer);
+}
+
+/**
+  Sample unit test that performs write past end of allocated buffer
+**/
+TEST (SanitizerTests, BufferOverflowWriteDeathTest) {
+  UINT8  *Pointer;
+
+  Pointer = (UINT8 *)AllocatePool (100);
+  ASSERT_NE (Pointer, (UINT8 *)NULL);
+
+  //
+  // Write past end of allocated buffer that should be caught by address sanitizer, log details, and exit
+  //
+  EXPECT_DEATH (Pointer[110] = 0, "ERROR: AddressSanitizer: heap-buffer-overflow");
+
+  FreePool (Pointer);
+}
+
+/**
+  Sample unit test that performs read before beginning of allocated buffer
+**/
+TEST (SanitizerTests, BufferUnderflowReadDeathTest) {
+  UINT8  *Pointer;
+  UINT8  Value;
+
+  Pointer = (UINT8 *)AllocatePool (100);
+  ASSERT_NE (Pointer, (UINT8 *)NULL);
+
+  //
+  // Read past end of allocated buffer that should be caught by address sanitizer, log details, and exit
+  //
+  EXPECT_DEATH (Value = Pointer[-10], "ERROR: AddressSanitizer: heap-buffer-overflow");
+  ASSERT_EQ (Value, Value);
+
+  FreePool (Pointer);
+}
+
+/**
+  Sample unit test that performs read from address 0 (NULL)
+**/
+TEST (SanitizerTests, NullPointerReadDeathTest) {
+  UINT8  Value;
+
+  //
+  // Read from address 0 should be caught by address sanitizer, log details, and exit
+  //
+  EXPECT_DEATH (Value = *(UINT8 *)(NULL), "ERROR: AddressSanitizer: ");
+  ASSERT_EQ (Value, Value);
+}
+
+/**
+  Sample unit test that performs write to address 0 (NULL)
+**/
+TEST (SanitizerTests, NullPointerWriteDeathTest) {
+  //
+  // Write to address 0 should be caught by address sanitizer, log details, and exit
+  //
+  EXPECT_DEATH (*(UINT8 *)(NULL) = 0, "ERROR: AddressSanitizer: ");
+}
+
+/**
+  Sample unit test that performs read from invalid address -1
+**/
+TEST (SanitizerTests, InvalidPointerReadDeathTest) {
+  UINT8  Value;
+
+  //
+  // Read from address -1 should be caught by address sanitizer, log details, and exit
+  //
+  EXPECT_DEATH (Value = *(UINT8 *)(-1), "ERROR: AddressSanitizer: ");
+  ASSERT_EQ (Value, Value);
+}
+
+/**
+  Sample unit test that performs write to invalid address -1
+**/
+TEST (SanitizerTests, InvalidPointerWriteDeathTest) {
+  //
+  // Write to address -1 should be caught by address sanitizer, log details, and exit
+  //
+  EXPECT_DEATH (*(UINT8 *)(-1) = 0, "ERROR: AddressSanitizer: ");
+}
+
+UINTN
+DivideWithNoParameterChecking (
+  UINTN  Dividend,
+  UINTN  Divisor
+  )
+{
+  //
+  // Perform integer division with no check for divide by zero
+  //
+  return (Dividend / Divisor);
+}
+
+/**
+  Sample unit test that performs a divide by 0
+**/
+TEST (SanitizerTests, DivideByZeroDeathTest) {
+  //
+  // Divide by 0 should be caught by address sanitizer, log details, and exit
+  //
+  EXPECT_DEATH (DivideWithNoParameterChecking (10, 0), "ERROR: AddressSanitizer: ");
+}
+
 int
 main (
   int   argc,
   char  *argv[]
   )
 {
   testing::InitGoogleTest (&argc, argv);
+  testing::FLAGS_gtest_catch_exceptions = 0;

```

Comment:
Moving forward we should be adding this flag to any GoogleTest that uses EXPECT_DEATH?

Reply:
This is a setting for the address sanitizer, not EXPECT_DEATH().  When GoogleTest is combined with the address sanitizer, all exceptions need to be handled by the address sanitizer so it can show details of the failure. This can be done globally with setting the env var GTEST_CATCH_EXCEPTIONS=0.  I have done this env var setting in this PR for Azure Pipelines CI agents.  And it is recommended that developers set this as well to always see full details.

Reply:
I have updated the Integration Instruction in the description of this PR with those details.  They also need to be added to the UnitTestFrameworkPkg ReadMe.